### PR TITLE
OCPBUGS-6683: add message about working with restricted, but not restricted-v2

### PR DESCRIPTION
**Pull request by stlaz**
This is a manual backport of https://github.com/openshift/apiserver-library-go/pull/89

/cc @deads2k @ibihim

### CODE REVIEW

**Comment by deads2k** [comment]
/approve

leaving lgtm with 

/assign @ibihim

File: pkg/securitycontextconstraints/sccadmission/admission_test.go [hunk 1]
```diff
@@ -1135,6 +1135,108 @@ func TestAdmitPreferNonmutatingWhenPossible(t *testing.T) {
...
+
+	tryRestrictedMessage := "fails to validate against the `restricted-v2` security context constraint, but would validate successfully against the `restricted`"
+
+	tests := map[string]struct {
+		oldPod            *coreapi.Pod
+		newPod            *coreapi.Pod
+		operation         admission.Operation
+		sccs              []*securityv1.SecurityContextConstraints
+		expectedMessage   string
+		notNotHaveMessage string
```
Comment (ibihim) [inline pkg/securitycontextconstraints/sccadmission/admission_test.go:1174]: Is this a typo?

**Pull request merged**